### ISSUE TITLE
Compilation error when importing enum from node modules in typescript

### ISSUE THREAD

**Issue by Pochwar** [symfony/webpack-encore#1060]
Hi,

I'm trying to use some enum from a library I made in a Symfony project.
The library is made with Vue and typescript, bundled with rollup.

The project is made Symfony 4.2, and the front with Vue 2 and typescript, Webpack Encore version is 1.6.1

The library is a dependency from my project, so I try to import the enum like this: `import { MyEnum } from 'myLibrary/src/enum/MyEnum';`

And the enum looks like this:
```javascript
// node_modules/myLibrary/src/enum/MyEnum.ts

export enum MyEnum {
  One = 'one',
  Two = 'two',
  Three = 'three'
}
```

But when I compile I got this error in console:
```
ERROR  Failed to compile with 1 errors                                                       4:37:05 PM

Error loading ./node_modules/myLibrary/src/enum/MyEnum.ts

 FIX  To process TypeScript files:
        1. Add Encore.enableTypeScriptLoader() to your webpack.config.js file.
```
And this error in browser:
```
Module parse failed: Unexpected token (4:7)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
> export enum MyEnum {
|   One = 'one',
|   Two = 'two'
```

I obviously already added `enableTypeScriptLoader()` in webpack.config.js.

If I create the same enum file into my project and import it, it works. And if I import interfaces from the same library, it works too.

I test my library in another project with VueCli and I got no error, so I'm pretty sure it comes from Webpack Encore.

Any idea how to solve this?

Thanks

**Comment by Kocal** [comment]
Hi!

It looks like `ts-loader` (used by Webpack Encore) has an option [`allowTsInNodeModules`](https://github.com/TypeStrong/ts-loader#allowtsinnodemodules) that is `false` by default.

What happens if you use the following configuration?
```js
Encore.enableTypeScriptLoader(function(options) {
  options.allowTsInNodeModules = true;
});
```

**Comment by Pochwar** [comment]
Hi thanks for your quick answer, but it doesn't solve the issue :/

Here's my webpack.config.js:

```
/* eslint-disable */
var Encore = require('@symfony/webpack-encore');
var path = require('path');

Encore
    // directory where compiled assets will be stored
    .setOutputPath('public/compiled/')
    // public path used by the web server to access the output path
    .setPublicPath('/compiled')
    // only needed for CDN's or sub-directory deploy
    //.setManifestKeyPrefix('build/')

    /*
     * ENTRY CONFIG
     *
     * Add 1 entry for each "page" of your app
     * (including one that's included on every page - e.g. "app")
     *
     * Each entry will result in one JavaScript file (e.g. app.js)
     * and one CSS file (e.g. app.css) if you JavaScript imports CSS.
     */
    // JS Front
    .addEntry('js/front', [
      './node_modules/bootstrap/dist/js/bootstrap.js',
      './assets/front/js/script.js',
    ])

    .addEntry('js/front/vue', './assets/front/vue/index.ts')

    // When enabled, Webpack "splits" your files into smaller pieces for greater optimization.
    //.splitEntryChunks()
    // will require an extra script tag for runtime.js
    // but, you probably want this, unless you're building a single-page app
    //.enableSingleRuntimeChunk()
    .disableSingleRuntimeChunk()

    /*
     * FEATURE CONFIG
     *
     * Enable & configure other features below. For a full
     * list of features, see:
     * https://symfony.com/doc/current/frontend.html#adding-more-features
     */
    .cleanupOutputBeforeBuild()
    .enableBuildNotifications()
    .enableSourceMaps(!Encore.isProduction())
    // enables hashed filenames (e.g. app.abc123.css)
    .enableVersioning(Encore.isProduction())

    .enableSassLoader((options) => {
      options.sourceMap = true;
      options.sassOptions = {
        outputStyle: options.outputStyle,
        includePaths: ['node_modules'],
        sourceComments: !Encore.isProduction(),
      };
      delete options.outputStyle;
    }, {})
    // CSS Front
    .addStyleEntry('css/front', ['./assets/front/scss/styles.scss'])
    .addStyleEntry(
      'css/admin', [
        './assets/admin/css/styles.css',
    ])

    // enable Vue
    .enableVueLoader()

    // enable TypeScript
    .enableTypeScriptLoader(function(options) {
      options.allowTsInNodeModules = true;
    })

    // PostCss for Tailwind
    .enablePostCssLoader()

    // Disable Firewall Hot Reloading
    .configureDevServerOptions((options) => {
      options.allowedHosts = 'all';
    })

    // ES Lint
    .addLoader({
      enforce: 'pre',
      test: /\.(js|vue)$/,
      loader: 'eslint-loader',
      exclude: [/node_modules/, /public/, /assets\/js/, /assets\/front\/js/],
      options: {
        fix: false,
        emitError: false,
        emitWarning: false,
      },
    })

    // Aliases
    .addAliases({
      '@': path.resolve(__dirname, 'assets/front/vue'),
      '@Img': path.resolve(__dirname, 'assets/front/img'),
      '@Animations': path.resolve(__dirname, 'assets/front/animations'),
      '@Style': path.resolve(__dirname, 'assets/front/scss'),
    })
;

module.exports = Encore.getWebpackConfig();
```

**Comment by floviolleau** [comment]
Hi @Pochwar,

Maybe you need to add your module in the tsconfig.json?
Try by using `import * as enum from 'myLibrary/src/enum/MyEnum';`

**Comment by Pochwar** [comment]
Hi @floviolleau

I always get the same error.
Do you declare the module in a `.d.ts` file in the library?

Do you have a `node_modules/whitelisted_module.ts` file or is it a typo? (=> `"include": ["node_modules/whitelisted_module"]`)

Thanks!

**Comment by Pochwar** [comment]
Hello, I finally find how to fix this.

First, it wasn't only an enum import issue, because I had the same error message when importing some Classes from .ts files in the library.

@Kocal  's solution was the first step, I added `allowTsInNodeModules` option in `enableTypeScriptLoader`'s options

```
Encore.enableTypeScriptLoader(function(options) {
  options.allowTsInNodeModules = true;
});
```

But this issue bring some info : https://github.com/symfony/webpack-encore/issues/186 : node_modules are excluded for this rule.

So, instead of deleting `rule.exclude` as mentioned in the issue, I modified the "exclude" regex to exclude node_modules but my own modules (that are in a '@myCompany' folder) like this

```
Encore.configureLoaderRule('typescript', (rule) => {
  rule.exclude = /node_modules\/(?!@myCompany)/;
});
```

and so, it works fine!
